plugins/woocommerce/src/Blocks/Utils/BlockTemplateUtils.php

```diff
@@ -59,6 +53,18 @@ class BlockTemplateUtils {
 	 */
 	const DEPRECATED_PLUGIN_SLUG = 'woocommerce';
 
+	/**
+	 * Returns the template matching the slug
+	 *
+	 * @param string $template_slug Slug of the template to retrieve.
+	 *
+	 * @return AbstractTemplate|AbstractTemplatePart|null
+	 */
+	public static function get_template( $template_slug ) {
+		$block_templates_registry = Package::container()->get( BlockTemplatesRegistry::class );
```

Comment:
My understanding is that ideally we should pass `BlockTemplatesRegistry` down from `Bootstrap.php`, instead of getting it directly here. But after trying with this approach (see b2d4a85fe8b4c066cf27cbc2c51a3bc81934caa2), it makes the code much more verbose as we need to pass the `BlockTemplatesRegistry` instance around in many methods. In a quick Slack chat with @dinhtungdu we thought it would be better to leave it like this for now.

Reply:
> My understanding is that ideally we should pass BlockTemplatesRegistry down from Bootstrap.php

Ideally, we should always do that. But static methods are stateless, we can't use a stateful approach (dependency injection) for stateless methods. Retrieving the instance from the container like this in static methods makes more sense to me, like how WooCommerce Core works with legacy code: https://github.com/woocommerce/woocommerce/tree/b05082ae874688db69a986cc03d2691b881a7163/plugins/woocommerce/includes#interacting-with-the-src-folder